src/main/java/efs/task/syntax/GuessNumberGame.java

```diff
@@ -13,10 +21,66 @@ public static void main(String[] args) {
     }
 
     public GuessNumberGame(String argument) {
-        //TODO: Implement the constructor
-    }
 
+        try {
+            this.M = Integer.parseInt(argument);
+        }
+        catch (NumberFormatException e) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new java.lang.IllegalArgumentException("Wrong format");
+        }
+        if (M > UsefulConstants.MAX_UPPER_BOUND || M < 1) {
+            System.out.println(UsefulConstants.WRONG_ARGUMENT);
+            throw new java.lang.IllegalArgumentException();
+        }
+
+        this.L = (int) ((Math.log(M) / Math.log(2)) + 1);
+        this.attempts = 0;
+        Random rand = new Random();
+        this.randomNumber = rand.nextInt(M) + 1;
+
+        progressBar = "[";
+        for(int i = 0; i < L; i++) {
+            progressBar = progressBar.concat(".");
+        }
+        progressBar = progressBar.concat("]");
+    }
     public void play() {
-        //TODO: Implement the method that executes the game session
+        Scanner input = new Scanner(System.in);
+        System.out.println("<1," + M + ">");
+        int playerGuess;
+
+        while(attempts < L) {
+            progressBar = progressBar.substring(0, attempts + 1) + "*" + progressBar.substring(attempts + 2);
+            System.out.println(progressBar);
```

Comment:
_clean code_ mimo, że jest to mało kodu, to pełni on odrębną funkcję od reszty tej metody. Warto byłoby wydzielić wypisywanie paska postępu do osobnej metody